src/builtins.c

```diff
@@ -259,6 +259,10 @@ JL_CALLABLE(jl_f_sizeof)
     jl_value_t *x = args[0];
     if (jl_is_unionall(x) || jl_is_uniontype(x)) {
         x = jl_unwrap_unionall(x);
+        size_t elsize = 0, al = 0;
+        int isinline = jl_islayout_inline(x, &elsize, &al);
```

Comment:
This will have the curious property that `T == S` will not imply `sizeof(T) == sizeof(S)`. I'm not sure if that's going to be an issue, but it does make be nervous. In particular, consider `Union{Tuple{Int64}, Tuple{Int8}}`, which doesn't include selector bits, vs. `Tuple{Union{Int64, Int8}}`, which throws an error.